cmd/multiFaExtract/multiFaExtract.go

```diff
@@ -5,54 +5,102 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
 	"github.com/vertgenlab/gonomics/common"
 	"github.com/vertgenlab/gonomics/fasta"
 	"log"
 )
 
-func multiFaExtract(infile string, outfile string, start int, end int, removeGaps bool) {
-	if !(start < end) {
-		log.Fatalf("Invalid arguments, start must be lower than end")
-	}
-	records := fasta.Read(infile)
+func multiFaExtract(s Settings) {
 	var ans []fasta.Fasta
-	for i := 0; i < len(records); i++ {
-		ans = append(ans, fasta.Extract(records[i], fasta.RefPosToAlnPos(records[0], start), fasta.RefPosToAlnPos(records[0], end), records[i].Name))
-	}
-	if removeGaps {
-		ans = fasta.RemoveGaps(ans)
+	records := fasta.Read(s.InFile)
+	if s.Bed == "" {
+		if !(s.Start < s.End) {
+			log.Fatalf("Invalid arguments, start must be lower than end")
+		}
+		ans = extractMultiHelper(records, s.Start, s.End)
+		if s.RemoveGaps {
+			ans = fasta.RemoveGaps(ans)
+		}
+		fasta.Write(s.OutFile, ans)
+	} else {
+		bedChan := bed.GoReadToChan(s.Bed)
+		for b := range bedChan {
+			ans = extractMultiHelper(records, b.ChromStart, b.ChromEnd)
+			if s.RemoveGaps {
+				ans = fasta.RemoveGaps(ans)
+			}
+			fasta.Write(fmt.Sprintf("%s.%v.%v.fa", b.Chrom, b.ChromStart, b.ChromEnd), ans)
+		}
 	}
+}
 
-	fasta.Write(outfile, ans)
+func extractMultiHelper(records []fasta.Fasta, start int, end int) []fasta.Fasta {
+	var ans = make([]fasta.Fasta, len(records))
+	for i := range records {
+		ans[i] = fasta.Extract(records[i], fasta.RefPosToAlnPos(records[0], start), fasta.RefPosToAlnPos(records[0], end), records[i].Name)
+	}
+	return ans
 }
 
 func usage() {
 	fmt.Print(
 		"multiFaExtract - Pull sub-sequence from multiple Fasta alignment for each entry. Uses reference indices, treating the first sequence as the reference.\n" +
 			"Usage:\n" +
 			"multiFaExtract multi.fa out.fa start end\n" +
+			"OR\n" +
+			"multiFaExtract -bed regions.bed multi.fa\n" +
```

Comment:
Does it make sense to merge these two commands into one, where the original/top command would just give a single bed entry?

Reply:
Are you suggesting we leave the usage the same but parse the start and end as a bed?

Reply:
you could just have one flag for `-region` that accepts either a bed file or an argument like `start:end`